plugins/source/aws/resources/services/iam/users_fetch.go

```diff
@@ -2,116 +2,67 @@ package iam
-	AccessKey2Active      bool      `csv:"access_key_2_active"`
-	AccessKey1LastRotated string    `csv:"access_key_1_last_rotated"`
-	AccessKey2LastRotated string    `csv:"access_key_2_last_rotated"`
-	Cert1Active           bool      `csv:"cert_1_active"`
-	Cert2Active           bool      `csv:"cert_2_active"`
-	Cert1LastRotated      string    `csv:"cert_1_last_rotated"`
-	Cert2LastRotated      string    `csv:"cert_2_last_rotated"`
-}
-
-type reportUsers []*reportUser
-
 const rootName = "<root_account>"
 
 func fetchIamUsers(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	return client.ListAndDetailResolver(ctx, meta, res, listUsers, userDetail)
+	return client.ListAndDetailResolver(ctx, meta, res, listUsers, getUserDetail)
 }
 
 func listUsers(ctx context.Context, meta schema.ClientMeta, detailChan chan<- interface{}) error {
-	report, err := getCredentialReport(ctx, meta)
-	if err != nil {
-		return err
-	}
-
-	for _, user := range report {
-		detailChan <- user
+	config := iam.ListUsersInput{}
+	c := meta.(*client.Client)
+	svc := c.Services().IAM
+	p := iam.NewListUsersPaginator(svc, &config)
+	for p.HasMorePages() {
+		response, err := p.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		for _, user := range response.Users {
+			detailChan <- user
+		}
 	}
 	return nil
 }
 
-func userDetail(ctx context.Context, meta schema.ClientMeta, resultsChan chan<- interface{}, errorChan chan<- error, listInfo interface{}) {
+func getUserDetail(ctx context.Context, meta schema.ClientMeta, resultsChan chan<- interface{}, errorChan chan<- error, listInfo interface{}) {
 	c := meta.(*client.Client)
 
-	reportUser := listInfo.(*reportUser)
-	if reportUser.ARN == c.AccountGlobalARN(client.IamService, "root") {
-		resultsChan <- UserWrapper{
-			User: types.User{
-				Arn:        aws.String(reportUser.ARN),
-				CreateDate: aws.Time(reportUser.UserCreationTime),
-				UserId:     aws.String("root"),
-				UserName:   aws.String(reportUser.User),
-			},
-			UserCreationTime:      reportUser.UserCreationTime,
-			PasswordStatus:        reportUser.PasswordStatus,
-			PasswordLastChanged:   reportUser.PasswordLastChanged,
-			PasswordNextRotation:  reportUser.PasswordNextRotation,
-			MfaActive:             reportUser.MfaActive,
-			AccessKey1Active:      reportUser.AccessKey1Active,
-			AccessKey2Active:      reportUser.AccessKey2Active,
-			AccessKey1LastRotated: reportUser.AccessKey1LastRotated,
-			AccessKey2LastRotated: reportUser.AccessKey2LastRotated,
-			Cert1Active:           reportUser.Cert1Active,
-			Cert2Active:           reportUser.Cert2Active,
-			Cert1LastRotated:      reportUser.Cert1LastRotated,
-			Cert2LastRotated:      reportUser.Cert2LastRotated,
-			reportUser:            reportUser,
-			isRoot:                true,
+	listUser := listInfo.(types.User)
+	arn := ""
+	if listUser.Arn != nil {
+		arn = *listUser.Arn
+	}
+	if arn == c.AccountGlobalARN(client.IamService, "root") {
```

Comment:
I don't think any of this is necessary anymore as `ListUser` never returns the root user